### DISCUSSION TITLE
Globbing images to resize them to a different directory

### DISCUSSION THREAD

**Discussion by Tom2112** [ImageMagick/ImageMagick#2736] [Help] [answered]
Image Magick says on their support pages that globbing files and processing them that way is more efficient/faster than doing the files individually. So I'm trying to figure out how to specify the output name of this command:

`magick -quality 60 *.jpg -resize "1920x1920>" C:\new_location\%d.jpg`

The above command processes the images perfectly, but names them 0.jpg, 1.jpg, 2.jpg, etc.

How do I get it to use the file's original name for the resized file, just in a different folder?

**Comment by snibgo** [comment, marked as answer]
I suggest using "magick mogrify". See https://imagemagick.org/script/mogrify.php . For example:
```
magick mogrify -quality 60 -path C:\new_location *.jpg
```
EDIT: Sorry, I put "magick mogrify" twice in the command.

**Comment by Tom2112** [comment]
Thank you for your quick reply!

That definitely works. I wonder why that same syntax isn't implemented in the regular magick command code...?

Anyway, THANK YOU!

**Reply by t246246** [reply]
if you want to use magick (convert), try

magick *.jpg -quality 60 -set filename:f "%t" c:/new_location/"[%filename:f].jpg"

**Comment by Tom2112** [comment]
Ugh! OK, now I'm trying to convert between formats:

magick *.jpg -quality 60 -define webp:image-hint=photo webp:image

Just creates one huge file named "image" which seems to be a slideshow of all the images converted to webp. How do I get separate webp files with the same names as the original files?

**Comment by fmw42** [comment]
`magick *.jpg -quality 60 -define webp:image-hint=photo +adjoin webp:image.webp
`
will produce image-0.webp image-1.webp ...

or

`magick *.jpg -set filename:fn "%t" -quality 60 -define webp:image-hint=photo +adjoin "webp:%[filename:fn].webp"`

provided you have enough RAM for all your JPGs

Otherwise, just use magick mogrify to convert a whole folder of images. See https://imagemagick.org/Usage/basics/#mogrify
